### ISSUE TITLE
Problems with duplicated transactions in genesis block

### ISSUE THREAD

**Issue by tmpolaczyk** [witnet/witnet-rust#1121]
If the `genesis_block.json` file contains a duplicated transaction, two transactions with the same hash will be inserted into the genesis block and their outputs will be stored only once in the UTXO set because one will overwrite the other.

Possible solutions:
* Check in `Block::genesis` that every transaction has a unique hash. This shouldn't be very costly as we already have to calculate the hash in order to build the merkle tree.
* Add a check for existing transactions in `UnspentOutputsPool::insert`. The node will panic with an error message. Even though that message does not include the problematic transaction, at least we know something is wrong.

Example problematic `genesis_block.json`:

```json
{
    "alloc": [
        [
            {
                "address": "twit1adgt8t2h3xnu358f76zxlph0urf2ev7cd78ggc",
                "value": "1000000000000000",
                "timelock": "0"
            }
        ],
        [
            {
                "address": "twit1adgt8t2h3xnu358f76zxlph0urf2ev7cd78ggc",
                "value": "1000000000000000",
                "timelock": "0"
            }
        ]
    ]
}
```

Note that this only affects transactions (not single outputs), so an address can have more than one output. Simple workarounds like changing the `value` or `timelock` field will change the transaction hash, so this issue is only an edge case. Storing all the outputs in a single transaction will also eliminate this problem.

**Comment by aesedepece** [comment]
This doesn't apply anymore, does it?

**Comment by aesedepece** [comment]
Ping.

**Comment by tmpolaczyk** [comment]
Well, the issue is still there, it's just that we decided not to fix it

**Comment by aesedepece** [comment]
Thanks, will close it and mark it as `wontfix`.
